### ISSUE TITLE
Datagrid manual override. 



### ISSUE THREAD

**Issue by cwinland** [stsrki/Blazorise#2041]
Problem: created a reusable component containing the data grid.  The fields are essentially the same,  but the data is different.  In addition, I use manual readdata on one usage and auto data on another when the data is much smaller.  The custom component allows me to configure some items but passing the parameters to the grid,  but I need to be able to turn on and off the manual vs auto. 

 When using blazor, I noticed that if I use readdata, even passing null,  it still uses manual data mode.   Ideally, I'd like to be able to override and use auto even if readdata is specified or use auto if readdata is null.   I can provide more specifics if needed. 

The work around was to either declare the grid twice in an if/else,  once when readdata has delegate and one when it doesn't.  That I couldn't do. So I decided to use afterrender to assign the readdata callback.  However,  this seems wrong. 

I also think onreaddata might be useful even in auto mode as an event to handle something. 

 Do you have any thoughts about this?

**Comment by stsrki** [comment]
I think an `if/else` is your only option for now. The thing is that as soon as you define ReadData it will go into the manual mode. There is no option to switch between manual and auto.

**Comment by David-Moreira** [comment]
I've also implemented both modes on mine.

Inheriting from Datagrid and have your class have a flag or something that defines the mode. Like you, I had to bind the ReadData on a livecycle method so it can be done dynamically I know it seems wrong, but I don't actually think there's another way without the else / if. 
I personally bind the ReadData on the OnParameteresSet method according to the given flag. 
I'm saying this from memory, if you eventually need more details I can go take a look at the implementation.
